x-pack/plugins/ingest_pipelines/public/application/components/pipeline_editor/__jest__/processors/circle.test.tsx

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { act } from 'react-dom/test-utils';
+import { setup, SetupResult, getProcessorValue } from './processor.helpers';
+
+const CIRCLE_TYPE = 'circle';
+
+describe('Processor: Circle', () => {
+  let onUpdate: jest.Mock;
+  let testBed: SetupResult;
+
+  beforeAll(() => {
+    jest.useFakeTimers();
+  });
+
+  afterAll(() => {
+    jest.useRealTimers();
+  });
+
+  beforeEach(async () => {
+    onUpdate = jest.fn();
+
+    await act(async () => {
+      testBed = await setup({
+        value: {
+          processors: [],
+        },
+        onFlyoutOpen: jest.fn(),
+        onUpdate,
+      });
+    });
+    testBed.component.update();
+    const {
+      actions: { addProcessor, addProcessorType },
+    } = testBed;
+    // Open the processor flyout
+    addProcessor();
+
+    // Add type (the other fields are not visible until a type is selected)
+    await addProcessorType(CIRCLE_TYPE);
+  });
+
+  test('prevents form submission if required fields are not provided', async () => {
+    const {
+      actions: { saveNewProcessor },
+      form,
+    } = testBed;
+
+    // Click submit button with only the type defined
+    await saveNewProcessor();
+
+    // Expect form error as "field" is required parameter
+    expect(form.getErrorsMessages()).toEqual([
```

Comment:
It looks like [`error_distance` should also be a required option](https://www.elastic.co/guide/en/elasticsearch/reference/master/ingest-circle-processor.html). I see the default value is set to `1` in the UI, which is why there is no error message here. But if you remove the value it sets it to `0` on save, rather than showing a field error. I don't know enough about the `error_distance` option to say whether this would be expected behavior or not. It may be worth reaching out to an ES engineer to ask though. It might be safer to show a field error, in which case we should open up a separate issue to address.

Reply:
See comment thread above. I opened https://github.com/elastic/kibana/issues/102147 to address.

Reply:
OK. So I went ahead and set the error_distance in the two tests where we are setting parameters. I see the comment in #102147 where you said you will update the behavior to validate when the parameter is not sent in the first test. So I will leave that to you and get this merged. @alisonelizabeth